Library/Homebrew/tap.rb

```diff
@@ -17,6 +17,9 @@ def self.clear_cache
   end
 
   def self.fetch(user, repo)
+    # we special case homebrew so users don't have to shift in a terminal
+    user = "Homebrew" if user == "homebrew"
+    repo = repo[9..-1] if repo.start_with? "homebrew-"
```

Comment:
I found regex's performance is far worse than substring. And it's a hot-spot here. Some benchmark:

```ruby
require "benchmark/ips"

def regex(input)
  input.sub(/^homebrew-/, "")
end

def substr(input)
  input = input[9..-1] if input.start_with? "homebrew-"
  input
end

def test(method)
  %w[
    homebrew-foo
    bar
  ].each { |s| send(method, s) }
end

Benchmark.ips do |x|
  x.report("use regex") { test :regex }
  x.report("use substr") { test :substr}
  x.compare!
end
```

While yields following result:
```
Calculating -------------------------------------
           use regex    30.447k i/100ms
          use substr    59.048k i/100ms
-------------------------------------------------
           use regex    395.658k (± 5.6%) i/s -      1.979M
          use substr    971.412k (± 5.3%) i/s -      4.901M

Comparison:
          use substr:   971411.6 i/s
           use regex:   395658.0 i/s - 2.46x slower
```